actions/loadconfig.py

```diff
@@ -1,29 +1,33 @@
 from lib.action import NapalmBaseAction
 
+from napalm.base.exceptions import MergeConfigException
+from napalm.base.exceptions import ReplaceConfigException
+
 
 class NapalmLoadConfig(NapalmBaseAction):
     """Load configuration into network device via NAPALM
     """
 
-    def run(self, config_file, method, inline_transfer, **std_kwargs):
-
-        if not method:
-            method = 'merge'
-        else:
-            method = method.lower()
-            if method not in ["merge", "replace"]:
-                raise ValueError(('{} is not a valid load method, use: '
-                                  'merge or replace').format(method))
-
-        with self.get_driver(**std_kwargs) as device:
-            # inline_transfer: If set it becomes True, else False
-            device.inline_transfer = inline_transfer
-
-            if method == "replace":
-                device.load_replace_candidate(filename=config_file)
-            else:
-                device.load_merge_candidate(filename=config_file)
-
-            device.commit_config()
+    def run(self, config_file, config_text, method, inline_transfer, **std_kwargs):
+        try:
+            if not config_file and not config_text:
+                raise ValueError('Specify either config_file or config_text')
+
+            with self.get_driver(**std_kwargs) as device:
+                # inline_transfer: If set it becomes True, else False
+                device.inline_transfer = inline_transfer
+
+                if method == "replace":
+                    device.load_replace_candidate(filename=config_file, config=config_text)
+                else:
+                    device.load_merge_candidate(filename=config_file, config=config_text)
+
+                device.commit_config()
+        except ValueError as e:
```

Comment:
https://github.com/StackStorm-Exchange/stackstorm-napalm/pull/34 moved away from catch-all exception handling, so I would prefer to not handle exceptions in the action unless there's something to do with the exception programmatically. If we can't actually **handle** the error, let's let the full stack trace bubble back up to the user.

The `ValueError` which is thrown when `config_file` and `config_text` is `None` is fine - that can stay in.

Reply:
Thank you for your review.
I understood the reason why you purposely not catch exception for telling detail of an exception to user. I'll fix it soon.